src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -683,22 +703,26 @@ internal IEnumerable<PackageManagerSearchElementViewModel> Search(string query)
             if (LastSync == null) return new List<PackageManagerSearchElementViewModel>();
 
             var canLogin = PackageManagerClientViewModel.AuthenticationManager.HasAuthProvider;
+            List<PackageManagerSearchElementViewModel> list = null;
 
             if (!String.IsNullOrEmpty(query))
             {
-                return
-                    SearchDictionary.Search(query)
-                        .Select(x => new PackageManagerSearchElementViewModel(x, canLogin))
-                        .Take(MaxNumSearchResults);
+                list = SearchDictionary.Search(query)
+                    .Select(x => new PackageManagerSearchElementViewModel(x, canLogin))
+                    .Take(MaxNumSearchResults).ToList();
             }
-
-            // with null query, don't show deprecated packages
-            var list =
-                LastSync.Where(x => !x.IsDeprecated)
+            else
+            {
+                // with null query, don't show deprecated packages
+                list = LastSync.Where(x => !x.IsDeprecated)
                     .Select(x => new PackageManagerSearchElementViewModel(x, canLogin)).ToList();
-            Sort(list, this.SortingKey);
-            return list;
+                Sort(list, this.SortingKey);
+            }
 
+            foreach (var x in list)
+                x.RequestShowFileDialog += OnRequestShowFileDialog;
+
+            return list;
```

Comment:
This is to add `OnRequestShowFileDialog` to each new instance of `PackageManagerSearchElementViewModel`.